statsmodels/discrete/tests/test_count_model.py

```diff
@@ -221,24 +221,54 @@ def setup_class(cls):
         model = sm.ZeroInflatedPoisson(cls.endog, exog)
         cls.res = model.fit(method='bfgs', maxiter=5000, maxfun=5000, disp=0)
 
+        cls.params_true = [mu_true,  0.05, nobs]
+
     def test_mean(self):
+        def compute_conf_interval_95(mu, prob_infl, nobs):
+            dispersion_factor = 1 + prob_infl * mu
+
+            # scalar variance of the mixture of zip
+            var = (dispersion_factor*(1-prob_infl)*mu).mean()
+            var += (((1-prob_infl)*mu)**2).mean()
+            var -= (((1-prob_infl)*mu).mean())**2
+            std = np.sqrt(var)
+            # Central limit theorem
+            conf_intv_95 = 2 * std / np.sqrt(nobs)
+            return conf_intv_95
+
+        conf_interval_95 = compute_conf_interval_95(*self.params_true)
         assert_allclose(self.res.predict().mean(), self.endog.mean(),
-                        atol=1e-2, rtol=1e-2)
+                        atol=conf_interval_95, rtol=0)
```

Comment:
A 95% CI seems to be far too wide. What is the tightest internal where this passes?

Reply:
On this PR: from parameters, we are creating a toy dataset with `.rvs()` and fitting on it, then we are comparing the fitted mean with the true mean. So the fit is heavily conditionned by the samples. I figured out that taking the CI would take into account that variability.

If you want stricter test, we can keep the current version which is the same test but comparing the mean of the toy dataset to the fitted mean. Any suggestion on how to fix the `atol` and `rtol`, so that it does not fail when changing the seed or the `nobs` ?